File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/getshapeicon.rb
module Asposediagramjava
  module GetShapeIcon
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Basic_Shapes.vsd")

        # get master
        master = diagram.getMasters().getMasterByName("Circle")
        
        # get byte array
        bytes = master.getIcon()
        
        # create an image file
        fos = Rjb::import('java.io.FileOutputStream').new(data_dir + "MyImage.png")
        
        # write byte array of the image
        fos.write(bytes)
        
        # close array
        fos.close()

        puts "Get shape icon, please check the output file."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/GetShapeIcon.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;

use java\io\FileOutputStream as FileOutputStream;
class GetShapeIcon{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing1.vsd");

        # get master
        $master = $diagram->getMasters()->getMasterByName("Circle");

        # get byte array
        $bytes = $master->getIcon();

        # create an image file
        $fos = new FileOutputStream($dataDir."star.png");

        # write byte array of the image
        $fos->write($bytes);

        # close array
        $fos->close();
        print "Get shape icon, please check the output file.".PHP_EOL;
    }
}